io-engine/src/rebuild/rebuild_job.rs

```diff
@@ -164,9 +165,19 @@ impl RebuildJob {
 
     /// Schedules the job to start in a future and returns a complete channel
     /// which can be waited on.
-    pub fn start(
+    pub async fn start(
         &self,
+        map: Option<RebuildMap>,
```

Comment:
Is this because the rebuild cannot be created with the map already?

Reply:
Rebuild map is optional, e.g. if a child wasn't sync, or if I/O logging is disabled.

Reply:
Ah right...